### PR TITLE
Date Range on Search and Filters

### PR DESCRIPTION
**Pull request by owen-sellner**
## Notion task link

[Date Range on Search and Filters](https://www.notion.so/uwblueprintexecs/Date-Range-on-Search-and-Filters-c9e63b4208e74371a46e1f3ca64fb50b)

## Implementation description
* Updated `SearchAndFilters`to function with only either a start date or end date.
* Added error validation to `SearchAndFilters` if the start date is after the end date
* Used a toast message to notify the user of the error

## Steps to test
1. Create new logs with a variety of dates
2. Verify that filtering works with no start date and no end date
3. Verify that filtering  works with only start date
4. Verify that filtering  works with only end date
5. Verify that filtering  works with both start date and end date
6. Check that the clear button works correctly

## Screenshot
image

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] If I have made API changes, I have updated the [REST API Docs](https://www.notion.so/uwblueprintexecs/REST-Endpoints-05ce60312bb943439dfda42bb1318536)
- [ ] IF I have made changes to the db/models, I have updated the [Data Models Page](https://www.notion.so/uwblueprintexecs/Data-Models-760f8aa06b244eb0842c079ad77987b0)
- [ ] I have updated other Docs as needed

### CODE REVIEW

**Comment by braydonwang** [comment]
Overall PR looks good! The only comment I have (that could be addressed in a follow-up PR) is should we consider not even changing the date if it's invalid? (if start date is after end date) Right now, if the user selects a start date that occurs after the end date, we get a toast message, but the start date's value still changes, which is probably not ideal

File: frontend/src/components/pages/HomePage/HomePage.tsx [hunk 1]
```diff
@@ -45,26 +45,28 @@ const HomePage = (): React.ReactElement => {
...
   const tableRef = useRef<HTMLDivElement>(null);
 
-  const formatDate = (date: Date) => {
-    const isoString = date.toISOString(); // Get ISO string, e.g., "2023-07-09T00:00:00.000Z"
-    const formattedDate = isoString.slice(0, 10); // Extract "YYYY-MM-DD" from the ISO string
-    return formattedDate;
+  const formatDate = (date: Date | undefined) => {
+    if (date !== undefined) {
+      return date
+        .toLocaleString("fr-CA", { timeZone: "America/Toronto" })
```
Comment (connor-bechthold) [inline frontend/src/components/pages/HomePage/HomePage.tsx:51]: I see you changed this to `.toLocaleString()` from `.toISOString()`. I think this is correct, but it's probably worth doing more testing on this once this in prod to ensure the dates aren't doing something weird (just making a note for the future)

File: frontend/src/components/pages/HomePage/HomePage.tsx [hunk 2]
```diff
@@ -45,26 +45,28 @@ const HomePage = (): React.ReactElement => {
   // Table reference
   const tableRef = useRef<HTMLDivElement>(null);
 
-  const formatDate = (date: Date) => {
-    const isoString = date.toISOString(); // Get ISO string, e.g., "2023-07-09T00:00:00.000Z"
-    const formattedDate = isoString.slice(0, 10); // Extract "YYYY-MM-DD" from the ISO string
-    return formattedDate;
+  const formatDate = (date: Date | undefined) => {
+    if (date !== undefined) {
```
Comment (connor-bechthold) [inline frontend/src/components/pages/HomePage/HomePage.tsx]: Nitpick but this can just be `if (date)`

**Review by connor-bechthold — APPROVED** [review]

**Comment by connor-bechthold** [comment]
> Overall PR looks good! The only comment I have (that could be addressed in a follow-up PR) is should we consider not even changing the date if it's invalid? (if start date is after end date) Right now, if the user selects a start date that occurs after the end date, we get a toast message, but the start date's value still changes, which is probably not ideal

@braydonwang Did you test this out on your end? I pulled this branch and tested for myself and the state doesn't update (which matches what the code reads)

File: frontend/src/components/pages/HomePage/SearchAndFilters.tsx [hunk 3]
```diff
@@ -79,6 +80,8 @@ const SearchAndFilters = ({
   const [userLabels, setUserLabels] = useState<UserLabel[]>();
   const [residentLabels, setResidentLabels] = useState<ResidentLabel[]>();
 
+  const newToast = CreateToast();
```
Comment (kevin-pierce) [inline frontend/src/components/pages/HomePage/SearchAndFilters.tsx]: nit (first of the season 😎): I think it's better if we name this `dateChangeToast` or something similar so the intention is clearer

Comment (owen-sellner) [reply frontend/src/components/pages/HomePage/SearchAndFilters.tsx]: first nit 🥳

**Review by phamkelly17 — APPROVED** [review]
lgtm

**Pull request merged**